solo/solo.py

```diff
@@ -108,63 +83,26 @@ def main():
         os.mkdir(args.out_dir)
 
     if args.reproducible_seed is not None:
-        torch.manual_seed(args.reproducible_seed)
-        np.random.seed(args.reproducible_seed)
+        scvi.settings.seed = args.reproducible_seed
+    else:
+        scvi.settings.seed = np.random.randint(10000)
+
     ##################################################
     # data
 
     # read loom/anndata
     data_ext = os.path.splitext(data_path)[-1]
     if data_ext == '.loom':
-        scvi_data = LoomDataset(data_path)
+        scvi_data = read_loom(data_path)
     elif data_ext == '.h5ad':
-        adata = anndata.read(data_path)
-        if issparse(adata.X):
-            adata.X = adata.X.todense()
-        scvi_data = AnnDatasetFromAnnData(adata)
-    elif os.path.isdir(data_path):
-        scvi_data = Dataset10X(save_path=data_path,
-                               measurement_names_column=1,
-                               dense=True)
-        cell_umi_depth = scvi_data.X.sum(axis=1)
-        fifth, ninetyfifth = np.percentile(cell_umi_depth, [5, 95])
-        min_cell_umi_depth = np.min(cell_umi_depth)
-        max_cell_umi_depth = np.max(cell_umi_depth)
-        if fifth * 10 < ninetyfifth:
-            print("""WARNING YOUR DATA HAS A WIDE RANGE OF CELL DEPTHS.
-            PLEASE MANUALLY REVIEW YOUR DATA""")
-        print(f"Min cell depth: {min_cell_umi_depth}, Max cell depth: {max_cell_umi_depth}")
+        scvi_data = read_h5ad(data_path)
     else:
         msg = f'{data_path} is not a recognized format.\n'
-        msg += 'must be one of {h5ad, loom, 10x directory}'
+        msg += 'must be one of {h5ad, loom}'
```

Comment:
you can easily add readers from scanpy to support 10x loading

Reply:
Is it fairly recent you depend on `scanpy`? Ya adding this should be no problem then

Reply:
well, we don't explicitly rely on scanpy, it's an optional req. The readers we have are directly imported from anndata, not scanpy. That said, it seems reasonable for this repo to depend on scanpy.